dbos/workflows_test.go

```diff
@@ -142,11 +148,11 @@ func TestWorkflowsRegistration(t *testing.T) {
 				result, err := handle.GetResult()
 				_, err2 := handle.GetResult()
 				if err2 == nil {
-					t.Fatal("Second call to GetResult should return an error")
+					return nil, fmt.Errorf("Second call to GetResult should return an error")
```

Comment:
`t.Fatal` terminates the goroutine, but this is an inner goroutine ran by the test goroutine itself. The resulting error is quite hard to parse _and_ the test goroutine will move on to its cleanup actions, potentially deadlocking w/o much information. So, as a rule, we should only call `t.Fatal` in the "top level" test goroutine.